packages/expo-gl/src/GLView.web.js

```diff
@@ -1,14 +1,196 @@
 // @flow
+import PropTypes from 'prop-types';
+import * as React from 'react';
+import { Dimensions, StyleSheet } from 'react-native';
+import { UnavailabilityError } from 'expo-errors';
+import type { SurfaceCreateEvent, SnapshotOptions } from './GLView.types';
 
-import React from 'react';
-import { Text, View } from 'react-native';
+type Props = {
+  /**
+   * Called when the OpenGL context is created, with the context object as a parameter. The context
+   * object has an API mirroring WebGL's WebGLRenderingContext.
+   */
+  onContextCreate?: (gl: *) => void,
+};
+
+/**
+ * A component that acts as an OpenGL render target
+ */
+let contextCache = {};
+
+function expoContext(gl: WebGLRenderingContext) {
+  gl.endFrameEXP = gl.endFrameEXP || function() {};
+  if (!gl.__expo_texImage2D) {
```

Comment:
Instead of saving __expo_texImage2D on the GL context object, save the original function in a local variable that we close over:
```js
const originalTexImage2D = gl.texImage2D;
gl.texImage2D = function texImage2D(...args) {
  ...
  return originalTexImage2D.call(gl, ...);
};
```